File: evaluation/coin/coin.py
import time
import pymc3 as pm
import numpy as np


def timer(f):
    t1 = time.time()
    x = f()
    t2 = time.time()
    print(t2-t1, "seconds")
    return x

def f():
    t = 0
    for _ in range(10000):
        t += np.random.beta(10,2)
    return t/10000

def f1():
    m = pm.Model()
    with m as model:
        theta = pm.Uniform('theta', 0, 1)
        obs = pm.Binomial('obs', n=10, p=theta, observed=9)
        step = pm.Metropolis()
        trace = pm.sample(1, chains=10000, step=step, tune=100)
        
    return np.mean(trace['theta'])

if __name__ == "__main__":
    print(timer(f1))

File: evaluation/coin/coin.js
var t0 = console.time("coin");

var coin = function() {
  var theta = uniform(0.0, 1.0);
  observe(Binomial({ n: 10, p: theta }), 9);
  return theta;
};

var inferred = Infer(
  { method: "MCMC", kernel: "MH", samples: 100000, burn: 100 },
  coin
);

var m = 0;
var n = 10000;
for (let x = 0; x < n; x++) {
  m = m + sample(inferred);
}
var mean = m / n;
console.log(mean);
var t1 = console.timeEnd("coin");
var x = sample(inferred);
console.log(x);
